src/WheelsAndGoods.DataAccess/Repositories/OrdersRepository.cs

```diff
@@ -51,9 +55,21 @@ public async Task<IReadOnlyCollection<Order>> GetOrders(FilterOrderModel filterO
                 query = query.Where(order => order.Price >= filterOrderModel.Price);
             }
 
-            return await query.OrderByDescending(order => order.CreatedAtUtc)
-                .AsNoTracking()
-                .ToArrayAsync();
+            query.OrderByDescending(order => order.CreatedAtUtc)
+                .Skip((paginationOrderModel.PageNumber - 1) * paginationOrderModel.PageSize)
+                .Take(paginationOrderModel.PageSize)
+                .AsNoTracking();
+
+            int totalCount = await query.CountAsync();
```

Comment:
totalCount should also respect the filter

Reply:
Explain pls. How totalCound should apply to the filter? I already take count from filtered query.

Reply:
Cool then. But there is another problem, looks like ordering and paging are not working now.